src/appchooserdialog.c

```diff
@@ -132,7 +132,9 @@ show_more (AppChooserDialog *dialog)
   gtk_window_set_resizable (GTK_WINDOW (dialog), TRUE);
   gtk_widget_hide (dialog->more_row);
 
-  for (i = INITIAL_LIST_SIZE - 1; dialog->choices[i]; i++)
+  g_assert (g_strv_length ((char **)dialog->choices) > INITIAL_LIST_SIZE);
```

Comment:
`g_return_if_fail()`, as mentioned, we want to avoid xdg-desktop-portal crashing for all users because of a programmer bug.

Reply:
As you prefer.